### DISCUSSION TITLE
FASTFARM nine turbine wind farm 7-9 turbine issues

### DISCUSSION THREAD

**Discussion by BAbaka626** [OpenFAST/openfast#2075] [Q&A]
Dear  @jjonkman 
I am using a 3*3 TSinflow - spar turbine with an incoming wind speed of 11.4, and the kaimal high precision wind farm used, 1-6 turbines work fine; 7-9 come up:
T7:FARM_UpdateStates:NearWakeCorrection:Wake model is not valid in the propeller-brake region,
i.e., Ct > 2.0.
T8:FARM_UpdateStates:NearWakeCorrection:Wake model is not valid in the propeller-brake region,
i.e., Ct > 2.0.
T9:FARM_UpdateStates:NearWakeCorrection:Wake model is not valid in the propeller-brake region,
i.e., Ct > 2.0.

Below is the operating box for the 9 turbine:
![屏幕截图 2024-03-06 103824](https://github.com/OpenFAST/openfast/assets/147319660/e755b8be-bd68-40a3-8885-13ca44a9bd14)
 
When I remove turbines 7-9, they run fine, and I'm using the same master file for all my fans, so I'm assuming that the fan parameters aren't what's causing the problem.
![image](https://github.com/OpenFAST/openfast/assets/147319660/470d0e77-d1a3-4cc7-b2bb-a934d03a6803)

![image](https://github.com/OpenFAST/openfast/assets/147319660/27cc3bac-8f10-437d-96ba-a72cf37a5221)

Can you tell me what went wrong and how to correct it?

yifan

**Comment by jjonkman** [comment]
Dear @BAbaka626,

If I understand correctly, you took the existing TSinflow r-test (https://github.com/OpenFAST/r-test/tree/main/glue-codes/fast-farm/TSinflow) and changed the OpenFAST models to be floating offshore atop a spar and increased from a 1x2 wind farm to 3x2 and 3x3 wind farms; is that correct?  Did you switch the OpenFAST models so that they match the r-test model of the NREL 5-MW baseline wind turbine atop the OC3-Hywind spar (https://github.com/OpenFAST/r-test/tree/main/glue-codes/openfast/5MW_OC3Spar_DLL_WTurb_WavesIrr) or something else?  What else did you change in the OpenFAST model?

I see that you are using the same OpenFAST model for all wind turbines, which differs from the TSinflow r-test, which uses distinct OpenFAST models because of the use of the DISCON controller, which must be distinct per turbine.  Did you swap out the DISCON controller with some other controller?  If not, your simulation results are likely corrupted as discussed in the online FAST.Farm documentation: https://openfast.readthedocs.io/en/main/source/user/fast.farm/InputFiles.html#openfast-input-files.

For both the 3x2 and 3x3 wind farm simulations, I see warnings about the SkewedWakeCorrection, which could be indicative of other problems with your simulation setup, but I'll let you respond to my first questions before speculating further.

Best regards,

**Comment by BAbaka626** [comment]
Dear @jjonkman 
thanks for your reply,

1. I download fast-farm main file from github . As you can see, I only changed :_fstf file_"a)2 wind turbines to 9   b) the beginning and the size of the high/low precision grid   ;To exclude the wind field problem, the windtype in the InflowWind has been setted to 1.. That's all.

![image](https://github.com/OpenFAST/openfast/assets/147319660/b6318c46-8ddc-4f01-9404-e0d2331e358c)
![image](https://github.com/OpenFAST/openfast/assets/147319660/ead49f03-e0f3-491a-8ad2-99c21cf7d376)

![image](https://github.com/OpenFAST/openfast/assets/147319660/84834e6d-558f-490b-bb37-3c927c8d3e0d)

2.As you can see, I take your advice just change the 7-9 wind turbines, the result gets better. It can run for 105s. However, the calculation is still not completed and errors similar to the previous one occur
![image](https://github.com/OpenFAST/openfast/assets/147319660/a680be0d-0fe6-486d-9f83-0453a08e16dd)

Best regards

**Reply by jjonkman** [reply]
Dear @BAbaka626,

If I understand correctly then, you are not modeling a spar, but rather using the _FFTest_WT#.fst_ OpenFAST models provided in the TSinflow r-test, which are models of the land-based NREL 5-MW baseline turbine.

Regardless, the problem I still believe you are running into is that you are using the exact same OpenFAST input files for turbines 1-6.  That is, you should use distinct `WT_FASTInFile` names, _FFTest_WT#.fst_ with # = 1,2,...,9, as well as distinct ServoDyn input files and distinct _DISCON.dll_ controllers.

Best regards,

**Comment by nixinonana** [comment]
Dear @jjonkman 
I had a similar problem with FASTfarm, who reported the following errors
`
T2:FARM_UpdateStates:FWrap_Increment:FAST_Solution:FAST_AdvanceStates:SolveOption2c_Inp2AD_SrvD:In
flowWind_CalcOutput:InflowWind_GetSpatialAverage:CalculateOutput:IfW_FlowField_GetVelAcc:Grid4DFie
ld_GetVel:Outside the grid bounds.`

I realize that this reporting error was once discussed among the next post, but I don't think it's the same problem because my calculations terminate when I get to 50s.
I used windtype=1 in inflowwind to exclude the wind field problem, my low precision grid range is X=[-500,1500], there are three wind turbines in x-direction at -504m,0m,504m.
https://github.com/OpenFAST/openfast/pull/1684


Best regards,

**Reply by nixinonana** [reply]
Incidentally, I use separate fst files and separate servodyn and dll control files for all my turbines.
The node where this error occurs is the one where I changed the size of the high precision grid and the low precision grid (the high precision grid was changed from 5m to 2m, and the low precision grid was changed from 10m to 4m, in the x, y, and z directions) but the computational sizes of the grid was unchanged

**Reply by jjonkman** [reply]
Dear @nixinonana,

I'm not sure I understand enough about your simulation set up to comment specfically.

Regardless, if this error is triggered 50-s into a simulation, the error implies that your high-resolution domain is not large enough to support the turbine deflections/displacements.  All aerodynamic analysis nodes distributed across the wind turbine, including any deflection/displacement of the floater/tower/yaw/blades, must remain within the associated high-resolution domain.

Best regards,
